c-dependencies/js-compute-runtime/builtins/json-web-key.cpp

```diff
@@ -0,0 +1,353 @@
+// TODO: remove these once the warnings are fixed
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Winvalid-offsetof"
+#pragma clang diagnostic ignored "-Wdeprecated-enum-enum-conversion"
+#include "js/ArrayBuffer.h"
+#include "js/Conversions.h"
+#include "js/ForOfIterator.h"
+#include "js/Object.h"
+#include "js/Promise.h"
+#include "js/experimental/TypedData.h"
+#include "jsapi.h"
+#include "jsfriendapi.h"
+#pragma clang diagnostic pop
+
+#include "js-compute-builtins.h"
+#include "json-web-key.h"
+
+namespace builtins {
+
+namespace {
+JS::Result<std::optional<std::string>>
+extractStringPropertyFromObject(JSContext *cx, JS::HandleObject object, std::string_view property) {
+  bool has_property;
+  if (!JS_HasProperty(cx, object, property.data(), &has_property)) {
+    return JS::Result<std::optional<std::string>>(JS::Error());
+  }
+  if (!has_property) {
+    return std::optional<std::string>(std::nullopt);
```

Comment:
Does this change work, or did you run into template argument deduction problems?
```suggestion
    return std::nullopt;
```

Reply:
This runs into the deduction problem you've mentioned -- 

>  no viable conversion from returned value of type 'const std::nullopt_t' to function return type 'JS::Result<std::optional<std::string>>